.github/workflows/generate-results.yml

```diff
@@ -0,0 +1,29 @@
+name: "Generate index.html"
+on:
+  push:
+    branches:    
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env: 
+      CI_COMMIT_MESSAGE: "[bot] update index.html"
+      CI_COMMIT_AUTHOR: github
+    steps:
+    - uses: actions/checkout@v2.3.2
+    - if: github.event.commits[0].message != env.CI_COMMIT_MESSAGE
+      run: |
+        bash generate-results.sh
+        git config --global user.name "${{ env.CI_COMMIT_AUTHOR }}"
+        git config --global user.email "${{ env.CI_COMMIT_AUTHOR }}@users.noreply.github.com"
+
+        git add -A
+        if git status | grep -q modified
```

Comment:
[for reviewer] support no change on index.html. Else raise error nothing to commit